### ISSUE TITLE
Is the default model implementation RAMP+?

### ISSUE THREAD

**Issue by unilight** [NKU-HLT/RAMP_MOS#1]
Hi @jiusansan222 , thank you for releasing the code!

I am wondering whether the model implementation is default set to be RAMP+?
For example I am looking at the datastore part: https://github.com/NKU-HLT/RAMP_MOS/blob/main/model/datastore.py#L27-L40
I think using the prior information is new in RAMP+. However, not all datasets have these prior info.
Wanted to know what I should do to run the original RAMP?

**Comment by jiusansan222** [comment]
Thank you for your question! Yes, the current implementation corresponds to the latest design from our journal version, where we leverage prior information—such as system-related factors that are commonly present across datasets and are relevant to quality. This prior knowledge is used to adjust the distance metric in order to assist with quality prediction, which is part of the RAMP+ approach.

If you would like to run the original RAMP without these adjustments, you can remove the code related to distance adjustment. This will allow you to revert to the original RAMP model.

Once the review and modifications are completed, I will release the journal version as soon as possible to provide a clearer understanding of the approach. Let me know if you need any further clarification or help!

**Comment by unilight** [comment]
@jiusansan222 Thank you for the clarifications!

So I have now finished the first stage training of RAMP, and I am doing the second stage training (the fusion net part). I am not sure what loss function to use, so I am using only a regerssion loss right now. The problem is that the loss does not seem to go down, and the validation numbers (MSE, LCC, etc.) do not move at all. I wonder if the same thing happened when you trained your fusion net?

**Comment by jiusansan222** [comment]
I don't think there is any issue with your training process. For the RAMP+ model, the first stage uses L1 loss, and the second stage uses both L1 loss and cross-entropy, which means the classification head is trained in the second stage. Although there are some small differences compared to the RAMP model, I don't think this is the cause of your problem. Perhaps you can try adjusting hyperparameters like lr, k, and top.
